File: bonus/meeting_rooms.java
public boolean canAttendMeetings(int[][] intervals)
{
    Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));

    for (int idx = 1; idx < intervals.length; idx++)
    {
        if (intervals[idx][0] < intervals[idx - 1][1])
        {
            return false;
        }
    }

    return true;
}

File: bonus/meeting_rooms.py
def canAttendMeetings(intervals):
    # Time complexity O(N * log N) due to the sort, N = len(intervals)
    # Python implements timsort which uses up to O(N) space

    intervals.sort()

    for idx in range(1, len(intervals)):
        if intervals[idx][0] < intervals[idx - 1][1]:
            return False

    return True
